src/Orleans/Statistics/CounterStatistic.cs

```diff
@@ -54,16 +55,21 @@ internal interface ICounter<out T> : ICounter
 
     internal class CounterStatistic : ICounter<long>
     {
-        [ThreadStatic]
-        private static List<long> perOrleansThreadCounters;
+        //Used to store long as reference type in static dictionary
+        private class ReferenceLong
+        {
+            public long Value;
+        }
+
         [ThreadStatic]
         private static bool isOrleansManagedThread;
+        [ThreadStatic]
+        private static List<ReferenceLong> allValuesFromSpecificThread;
```

Comment:
maybe rename to `allStatisticsFromSpecificThread` and `specificStatisticFromAllThreads`